app/javascript/react/screens/App/Overview/screens/MappingWizard/components/MappingWizardNetworksStep/components/NetworksStepForm/NetworksStepForm.js

```diff
@@ -285,60 +287,46 @@ class NetworksStepForm extends React.Component {
     const { selectedNode } = this.state;
     const isTargetNetwork = selectedNode.nodes;
 
-    if (isTargetNetwork) {
-      const updatedMappings = networksStepMappings
-        .map(targetClusterWithNetworksMappings => {
-          const {
-            nodes: networksMappings,
-            ...targetCluster
-          } = targetClusterWithNetworksMappings;
-          const updatedNetworksMappings = networksMappings.filter(
-            targetNetworkWithSourceNetworks =>
-              targetNetworkWithSourceNetworks.id !== selectedNode.id
-          );
-          return updatedNetworksMappings.length === 0
-            ? undefined
-            : {
-                ...targetCluster,
-                nodes: updatedNetworksMappings
-              };
-        })
-        .filter(item => item !== undefined);
-      onChange(updatedMappings);
-    } else {
-      const updatedMappings = networksStepMappings
-        .map(targetClusterWithNetworksMappings => {
-          const {
-            nodes: networksMappings,
-            ...targetCluster
-          } = targetClusterWithNetworksMappings;
-          const updatedNetworksMappings = networksMappings
-            .map(networksMapping => {
-              const {
-                nodes: sourceNetworks,
-                ...targetNetwork
-              } = networksMapping;
-              const updatedSourceNetworks = sourceNetworks.filter(
-                sourceNetwork => sourceNetwork.uid_ems !== selectedNode.uid_ems
-              );
-              return updatedSourceNetworks.length === 0
-                ? undefined
-                : {
-                    ...targetNetwork,
-                    nodes: updatedSourceNetworks
-                  };
-            })
-            .filter(item => item !== undefined);
-          return updatedNetworksMappings.length === 0
-            ? undefined
-            : {
-                ...targetCluster,
-                nodes: updatedNetworksMappings
-              };
-        })
-        .filter(item => item !== undefined);
-      onChange(updatedMappings);
-    }
+    // *********************
+    // NETWORKS STEP MAPPING
+    // *********************
+    // Target Cluster
+    // --> Target Network
+    // ----> Source network grouping(s)
+
+    // ****************
+    // NETWORKS MAPPING
+    // ****************
+    // Target Network
+    // --> Source network grouping(s)
+
+    const updatedMappings = isTargetNetwork
+      ? networksStepMappings
+          .map(networksStepMapping =>
+            removeTargetNetwork(networksStepMapping, selectedNode)
+          )
+          .filter(item => item !== null)
+      : networksStepMappings
+          .map(networksStepMapping => {
```

Comment:
while it can be premature optimization, it feels like we are looping over the mapping many many times?

Reply:
Hi @ohadlevy, thanks for taking a look!

Are you referring to the fact that I am `map`ping, and then following that with `filter`?  I will clean this up